src/osd/osd_types.cc

```diff
@@ -4752,13 +4752,15 @@ void object_info_t::encode(bufferlist& bl) const
   ::encode(local_mtime, bl);
   ::encode(data_digest, bl);
   ::encode(omap_digest, bl);
+  ::encode(expected_object_size, bl);
+  ::encode(expected_write_size, bl);
```

Comment:
Woah, I notice that this updates the encoding to version 16, but it doesn't match the one used in https://github.com/ceph/ceph/pull/9452/files (which includes an alloc_hint_flags value here). That's a big no-no — it means when the OSD process upgrades, it will try and decode this and crash when the bufferlist doesn't contain all the data it expects!

I'm not sure if it suffices to just encode an empty 0-value here or not.